grow/cache/podcache.py

```diff
@@ -105,7 +106,7 @@ def _write_json(self, path, obj):
         output = json.dumps(
             obj, cls=json_encoder.GrowJSONEncoder,
             sort_keys=True, indent=2, separators=(',', ': '))
-        temp_path = '{}.tmp'.format(path)
+        _, temp_path = tempfile.mkstemp()
```

Comment:
the first return is an open file descriptor, can you use that to write the file directly instead of using the pod's write_file wrapper?

something like:

```
fp, temp_path = tempfile.mkstemp()
fp.write(output)
fp.close()
# then move the file
```